### PR TITLE
handle *all* exceptions, capture stack trace

### PR DESCRIPTION
**Pull request by robashton**
**Description of the change**

Clearly and concisely describe the purpose of the pull request. If this PR relates to an existing issue or change proposal, please link to it. Include any other background context that would help reviewers understand the motivation for this PR.

---

**Checklist:**

- [ ] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [ ] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

File: src/Effect/Exception.erl [hunk 1]
```diff
@@ -5,19 +5,19 @@ showErrorImpl(E) -> message(E).
...
+stackImpl(Just, _Nothing,{_, _, S}) -> Just(unicode:characters_to_binary(io_lib:format("~p", [S]), utf8)).
 
 throwException(E) -> fun() ->
   erlang:error(E)
 end.
 
 catchException(C,T) -> fun () ->
   try T()
-  catch error:Reason -> (C(Reason))()
+  catch Class:Reason:Stacktrace -> (C({Class, Reason, Stacktrace}))()
```
Comment (nwolverson) [inline src/Effect/Exception.erl:21]: This implies OTP21+ doesn't it?

Comment (robashton) [reply src/Effect/Exception.erl:21]: For sure, I was being bold seeing as we are on 23 or 24 or something now

Comment (robashton) [reply src/Effect/Exception.erl:21]: I think it can be ifdefd if we care

Comment (nwolverson) [reply src/Effect/Exception.erl:21]: Hmm, there's no OTP version support policy on `purerl` just now and there probably should be. The old compiler fork used to be 19+, OTP21 is 3+ years old at this point, and one can certainly argue that if one is more risk averse when it comes to upgrades they wouldn't be using `purerl` right now.

**Pull request merged**